cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -214,15 +207,27 @@ if (NOT THRIFT_FOUND)
                         "-DWITH_STATIC_LIB=ON"
                         )
   if (MSVC)
-    set (THRIFT_CMAKE_ARGS "-DLIBEVENT_ROOT=${CMAKE_CURRENT_BINARY_DIR}/thrift_ep-prefix/src/thrift_ep/thirdparty/src/Libevent-release-2.1.7-rc"
-                           "-DFLEX_EXECUTABLE=${CMAKE_CURRENT_BINARY_DIR}/thrift_ep-prefix/src/thrift_ep/thirdparty/dist/winflexbison/win_flex.exe"
-                           "-DBISON_EXECUTABLE=${CMAKE_CURRENT_BINARY_DIR}/thrift_ep-prefix/src/thrift_ep/thirdparty/dist/winflexbison/win_bison.exe"
-                           "-DZLIB_INCLUDE_DIR=${ZLIB_INCLUDE_DIRS}"
-                           "-DZLIB_LIBRARY=${ZLIB_STATIC_LIB}"
-                           "-DWITH_SHARED_LIB=OFF"
-                           "-DWITH_PLUGIN=OFF"
-                           ${THRIFT_CMAKE_ARGS})
-    set (THRIFT_DEPENDENCIES zlib_ep)
+    set(WINFLEXBISON_VERSION 2.4.9)
+
+    set(WINFLEXBISON_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/winflexbison_ep/src/winflexbison_ep-install")
+    ExternalProject_Add(winflexbison_ep
+      URL https://github.com/lexxmark/winflexbison/releases/download/v.${WINFLEXBISON_VERSION}/win_flex_bison-${WINFLEXBISON_VERSION}.zip
+      URL_HASH MD5=a2e979ea9928fbf8567e995e9c0df765
+      SOURCE_DIR ${WINFLEXBISON_PREFIX}
+      CONFIGURE_COMMAND "" BUILD_COMMAND "" INSTALL_COMMAND "")
+    set(THRIFT_DEPENDENCIES ${THRIFT_DEPENDENCIES} winflexbison_ep)
+
+    set(THRIFT_CMAKE_ARGS "-DLIBEVENT_ROOT=${CMAKE_CURRENT_BINARY_DIR}/thrift_ep-prefix/src/thrift_ep/thirdparty/src/Libevent-release-2.1.7-rc"
```

Comment:
It might make sense to move definition of `LIBEVENT_VERSION` somewhere above and use it to set the version of libevent in path `/thrift_ep-prefix/src/thrift_ep/thirdparty/src/Libevent-release-2.1.7-rc`